### PR TITLE
Add diamond exercise

### PR DESCRIPTION
**Pull request by jimmbraddock**
I can't come up normal tests. Easest is to check summary diamond shape.

### CODE REVIEW

File: exercises/diamond/diamond.exs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+defmodule Diamond do
+  @doc """
+  List the allergies for which the corresponding flag bit is true.
```
Comment (dalexj) [inline exercises/diamond/diamond.exs]: forgot to change this

**Comment by dalexj** [comment]
I think for tests it would be best to have only 3-4 tests that look like this one, but for different letters:

```elixir
  @tag :pending
  test "letters form a diamond shape" do
    shape = Diamond.build_shape(?E)
    assert shape == """
\s   A    
\s  B B   
\s C   C  
\sD     D 
 E       E
\sD     D 
\s C   C  
\s  B B   
\s   A    
    """
  end
```

**Comment by jimmbraddock** [comment]
I used `\s`, because leading whitespace disappear and was `...A...\n` instead of `...A...\n`. Even Markdown removes the leading whitespaces. Or I can replace whitespace to another symbol. Or this problem of my editor.

**Comment by parkerl** [comment]
@jimmbraddock could you rebase master onto your branch? Thanks!

**Comment by jimmbraddock** [comment]
@parkerl , I rebased.

**Comment by parkerl** [comment]
Thanks 🎉

**Pull request merged**